test/ssttpcalculator/CalculatorServiceMaximumDurationInMonthsSpec.scala

```diff
@@ -0,0 +1,205 @@
+        debit("POA1", "2017-04-05", 2000, "2018-01-31"),
+        debit("POA2", "2017-04-05", 3000, "2018-01-31")),
+      returns = List(return2016, return2017)), 11)
+  )
+
+  private val scenariosBasedAroundLateFilers = Table(
+    ("todayDate", "self assessment", "expected answer"),
+    //Customer 20
+    //todo I am assuming the old charges will not appear in des?
+    (startDate.withMonth(7).withDayOfMonth(1), returnsAndDebits.copy(
+      debits  = List(
+        debit("BCD", "2017-04-05", 0, "2017-07-31"),
+        debit("Late payment penalty", "2015-04-05", 1000, "2017-04-28")),
+      returns = List(return2016, return2017)), 11),
+    //Customer 21
+    //todo will the Late payment penalty appear as two paymeant
+    (startDate.withMonth(4).withDayOfMonth(20), returnsAndDebits.copy(
+      debits  = List(
+        debit("BCD", "2017-04-05", 0, "2017-07-31"),
+        debit("Late payment penalty", "2016-04-05", 1000, "2017-04-28"),
+        debit("POA2", "2016-04-05", 5000, "2017-07-17")),
+      returns = List(return2016, return2017)), 11),
+    //Customer 22
+    (startDate.withMonth(3).withDayOfMonth(20), returnsAndDebits.copy(
+      debits  = List(
+        debit("Late payment penalty", "2016-04-05", 100, "2017-03-25")),
+      returns = List(return2016, return2017)), 11),
+    //Customer 24
+    (startDate.withMonth(3).withDayOfMonth(20), returnsAndDebits.copy(
+      debits  = List(
+        debit("POA2", "2016-04-05", 3000, "2017-07-31"),
+        debit("Late payment penalty", "2016-04-05", 100, "2017-03-17")),
+      returns = List(return2016, return2017)), 11),
+    //Customer 25
+    (startDate.withMonth(11).withDayOfMonth(20), returnsAndDebits.copy(
+      debits  = List(
+        debit("Late payment penalty", "2016-04-05", 100, "2016-12-16"),
+        debit("BCD", "2016-04-05", 50, "2017-01-31"),
+        debit("POA1", "2016-04-05", 1100, "2017-01-31"),
+        debit("POA2", "2016-04-05", 1100, "2017-01-31")),
+      returns = List(return2016, return2017)), 11)
+  )
+
+  private val scenariosBasedCustom = Table(
+    ("todayDate", "self assessment", "expected answer"),
+    (startDate.withMonth(12).withDayOfMonth(18), returnsAndDebits.copy(
+      debits  = List(
+        debit("IN1", "2018-04-05", 30, "2018-01-31"),
+        debit("IN2", "2018-04-05", 500, "2018-07-31")),
+      returns = List(return2017.copy(receivedDate = Some(LocalDate.of(2017, NOVEMBER, 24))), return2018)), 11))
+
+  //todo ask ela about 11 13 15
+  "maximumDurationInMonths for" should {
+    TableDrivenPropertyChecks.forAll(scenariosBasedAroundJanuaryDuedate) { (startDate: LocalDate, returnsAndDebits: ReturnsAndDebits, answer: Int) =>
+      s"scenarios Based Around January Due date return the correct number of months($answer) for start date : $startDate and SelfAssessmentDetails $returnsAndDebits" in {
+        val result: Int = maximumDurationInMonths(returnsAndDebits, startDate)
+        assert(result == answer)
```

Comment:
We'd rather use matchers so the above would look like:

`result shouldBe answer`

or even better after small renamings:

`result shouldBe expectedResult`

Reply:
Yeah this spec is a bit unrefined - I just renamed it here...